keps/sig-api-machinery/20190325-unions.md

```diff
@@ -162,16 +165,30 @@ type Union2 struct {
 type InlinedUnion struct {
 	Name string `json:"name"`
 
-	// +unionDeprecated
+	// +unionMemberDeprecated
 	// +optional
 	Field1 *int `json:"field1,omitempty"`
-	// +unionDeprecated
+	// +unionMemberDeprecated
 	// +optional
 	Field2 *int `json:"field2,omitempty"`
 
 	Union  `json:",inline"`
 	Union2 `json:",inline"`
 }
+
+// This uses a deducible discriminator.
+// +union
+type Deprecated struct {
+	// +discriminator
```

Comment:
unionDiscriminator?

Reply:
I'd really like to keep that name as short as possible, I don't think it's confusing.

Reply:
Then you need to fix the "unionDiscriminator"s above?